mpf/platforms/virtual.py

```diff
@@ -311,23 +320,41 @@ class VirtualSegmentDisplay(SegmentDisplayPlatformInterface):
 
     """Virtual segment display."""
 
-    __slots__ = ["text", "flashing", "colors"]
+    __slots__ = ["text", "flashing", "platform_options", "colors", "machine", "post_update_events"]
 
-    def __init__(self, number) -> None:
+    def __init__(self, number, platform_options, machine) -> None:
         """Initialise virtual segment display."""
         super().__init__(number)
+        self.machine = machine
```

Comment:
I assume this only works when you use the virtual platform, right?

What about this: A plugin (or device) just registers a handler on the segment_display device. This way we could mirror any segment display to MC. That way it would also work with real hardware attached.

Reply:
I like that idea, although I could use a little help in how this would work. What would trigger the plugin getting loaded (I suppose some config option relating to the segment display)? The plugin would need to know the bcp connection name and then generate the proper triggers when the segment display text or other parameters were updated (hook into the segment_display set_text and set_color methods?).

Reply:
I think I have an idea of how to implement this after studying the other mpf plugins. I'll give it a shot. Thanks.